lib/stella/searchable.rb

```diff
@@ -87,7 +88,12 @@ def index_slug
 
       # sets up mappings and settings for index
       def searchable(settings = Stella::Analysis::DEFAULT_SETTINGS, &block)
+        def field(field, opts = {})
+          es_field(field, opts)
+        end
```

Comment:
Yes, that's pretty much what I did. I extracted a `Stella::Parser` class which takes a reference to the model and moved the field parsing code in there. I also added a spec to ensure no namespace clashes occur at the model class level. Have a look.